Client/Frontend/Browser/UserScriptManager.swift

```diff
@@ -187,388 +41,165 @@ class UserScriptManager {
-                        in: .page)
-  }()
-
-  private var walletEthProviderJS: String?
-    
-  public func reloadUserScripts() {
-    tab?.webView?.configuration.userContentController.do {
-      $0.removeAllUserScripts()
-      // This has to be added before `packedUserScripts` because some scripts do
-      // rewarding even if the request is blocked.
-      if isRequestBlockingEnabled, let script = requestBlockingUserScript {
-        $0.addUserScript(script)
-      }
-      
-      self.packedUserScripts.forEach($0.addUserScript)
+    webView.configuration.userContentController.do { scriptController in
+      scriptController.removeAllUserScripts()
       
-      if isCookieBlockingEnabled, let script = cookieControlUserScript {
-        $0.addUserScript(script)
+      // Inject all base scripts
+      self.baseScripts.forEach {
+        scriptController.addUserScript($0)
       }
-
-      if let script = resourceDownloadManagerUserScript {
-        $0.addUserScript(script)
-      }
-
-      if let script = WindowRenderHelperScript {
-        $0.addUserScript(script)
-      }
-
-      if let script = FullscreenHelperScript {
-        $0.addUserScript(script)
-      }
-
-      if UIDevice.isIpad, isWebCompatibilityMediaSourceAPIEnabled, let script = PlaylistSwizzlerScript {
-        $0.addUserScript(script)
-      }
-
-      if isPlaylistEnabled, let script = PlaylistHelperScript {
-        $0.addUserScript(script)
+      
+      // Inject specifically RequestBlocking BEFORE other scripts
+      // this is because it needs to hook requests before RewardsReporting
+      if scripts.contains(.requestBlocking), let script = self.dynamicScripts[.requestBlocking] {
+        scripts.remove(.requestBlocking)
+        scriptController.addUserScript(script)
       }
-
-      if isMediaBackgroundPlaybackEnabled, let script = MediaBackgroundingScript {
-        $0.addUserScript(script)
+      
+      // Inject all static scripts
+      self.staticScripts.forEach {
+        scriptController.addUserScript($0)
       }
-
-      if isNightModeEnabled, let script = NightModeScript {
-        $0.addUserScript(script)
+      
+      // Inject all scripts that are dynamic, but always enabled
+      self.dynamicScripts.filter({ self.alwaysEnabledScripts.contains($0.key) }).forEach {
+        scriptController.addUserScript($0.value)
       }
       
-      if isDeAMPEnabled, let script = deAMPUserScript {
-        $0.addUserScript(script)
+      // Inject all optional scripts
+      self.dynamicScripts.filter({ scripts.contains($0.key) }).forEach {
+        scriptController.addUserScript($0.value)
       }
+    }
+  }
+  
+  // TODO: Get rid of this OR refactor wallet and domain scripts
+  func loadCustomScripts(into tab: Tab, userScripts: Set<ScriptType>, customScripts: Set<UserScriptType>, walletEthProviderScript: WKUserScript?) {
+    guard let webView = tab.webView else {
+      assertionFailure("Injecting Scripts into a Tab that has no WebView")
```

Comment:
Enabling night mode fails here with a crash

Reply:
@stoletheminerals is this still happening? I can't reproduce it :)

Reply:
Can't reproduce it anymore :expressionless: